x-pack/plugins/enterprise_search/server/lib/ml/fetch_ml_models.ts

```diff
@@ -147,22 +146,22 @@ const getModel = (modelConfig: MlTrainedModelConfig, modelStats?: MlTrainedModel
   return model;
 };
 
-const enrichModelWithDownloadStatus = async (
-  model: MlModel,
-  trainedModelsProvider: MlTrainedModels
-) => {
-  const modelConfigWithDefinitionStatus = await trainedModelsProvider.getTrainedModels({
-    model_id: model.modelId,
-    include: 'definition_status',
-  });
-
-  if (modelConfigWithDefinitionStatus && modelConfigWithDefinitionStatus.count > 0) {
-    // We're using NotDeployed for downloaded models. Downloaded is also a valid status, but we want to have the same
-    // status badge as for 3rd party models.
-    model.deploymentState = modelConfigWithDefinitionStatus.trained_model_configs[0].fully_defined
-      ? MlModelDeploymentState.NotDeployed
-      : MlModelDeploymentState.Downloading;
-  }
+const enrichModelWithDownloadStatus = (model: MlModel, trainedModelsProvider: MlTrainedModels) => {
+  return trainedModelsProvider
+    .getTrainedModels({
+      model_id: model.modelId,
+      include: 'definition_status',
+    })
+    .then((modelConfigWithDefinitionStatus) => {
```

Comment:
Chaining the handling callback to the promise instead of resolving it here (the processing logic is unchanged). The resolution happens one level above in the call stack for all promises.